pandas/tests/util/test_deprecate_nonkeyword_arguments.py

```diff
@@ -0,0 +1,99 @@
+"""
+Tests for the `deprecate_nonkeyword_arguments` decorator
+"""
+
+from pandas.util._decorators import deprecate_nonkeyword_arguments
+
+import pandas.util.testing as tm
+
+
+@deprecate_nonkeyword_arguments(version="1.1", allowed_args=["a", "b"])
+def f(a, b=0, c=0, d=0):
+    """
+    Sum of one to four numbers, the two last arguments
+    should be given as keyword arguments only
+    """
+    return a + b + c + d
+
+
+def test_one_arguments():
+    """
+    Check whether no future warning is produced if one
+    positional argument given.
+    """
+    assert f(19) == 19
```

Comment:
Need to assert that there's no warnings where relevant. Our CI sets things up so it's checked, but we shouldn't rely on that.

Reply:
Ok, I used `tm.assert_produces_warning(None)` just like in pandas/tests/io/json/test_deprecated_kwargs.py. Is it the proper way?

Reply:
I think so.